apps/system/js/statusbar.js

```diff
@@ -161,6 +91,7 @@ var StatusBar = {
     window.addEventListener('iconshown', this);
     window.addEventListener('iconhidden', this);
     window.addEventListener('iconchanged', this);
+    window.addEventListener('iconrendered', this);
```

Comment:
Well, I think we have a dependency issue here.
Doing this needs us to load and start statusbar before any icon is rendered, otherwise we will lose an icon which is already put to the statusbar before it starts to listen to rendered event.

Here is my idea to fix it:
1. When statusbar is starting, query all the current 'statusbar-icon' and try to find the icon instance by some reference.
2. Use Service.request in BaseIcon to draw the icon before Statusbar is inited. This will change the render function back to async again

I personally prefer 2. It makes more sense because semantically each module is requesting to render their icon to statusbar instead of putting the icon when the module is started without knowing statusbar is ready or not.

Thought?